javalib/src/main/scala/java/util/regex/cre2.scala

```diff
@@ -1,167 +0,0 @@
-package java.util.regex
-
-import scalanative.native._
-import scalanative.libc.stdlib._
-
-import java.nio.charset.Charset
-
-// Syntax guide: https://github.com/google/re2/wiki/Syntax
-// re2 c wrapper: https://github.com/marcomaggi/cre2
-@link("re2")
-@extern
-object cre2 {
```

Comment:
Please also delete `nativelib/src/main/resources/optional/re2.{h, cpp}`.

Reply:
A few more places where c-based re2 is referenced:
1. scripts/scala-native.nix needs to drop `re2` line.
2. scripts/travis_setup.sh we **can't** drop it there as this script used by 0.3.x users as well.
3. docs/lib/javalib.rst needs to drop all lines that contain `cre2`.
4. docs/user/setup.rst needs to drop `re2` in all setup instructions.

Reply:
Done. travis_setup.sh not touched.  Sorry I missed so many places.

Reply:
I have no way to test scala-native.nix. Will Travis CI do that or can you suggest a way?
I would  rest easier knowing I did not inflict breakage on someone else.  

I will test the  *.rst changes on my machine so they get exercised before the
ramp up to release.